### PR TITLE
Optional query param

### PR DESCRIPTION
**Pull request by mapingo**

### CODE REVIEW

File: adapters/adapters-test-utils/src/main/java/uk/gov/justice/services/adapters/test/utils/dispatcher/SynchronousRecordingDispatcher.java [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
 
+import javax.inject.Singleton;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
 
-import javax.inject.Singleton;
 
 @Singleton
 public class SynchronousRecordingDispatcher extends BasicRecordingDispatcher implements SynchronousDispatcher {
```
Comment (purple52) [inline adapters/adapters-test-utils/src/main/java/uk/gov/justice/services/adapters/test/utils/dispatcher/SynchronousRecordingDispatcher.java]: Is everyone using the correct coding style settings?

File: adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/application/CommonProviders.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Configuration class containing common JAX-RS providers to be included in service level JAX-RS application classes.
+ */
+public class CommonProviders {
+
+    public Set<Class<?>> providers() {
+        HashSet<Class<?>> classes = new HashSet<>();
```
Comment (purple52) [inline adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/application/CommonProviders.java]: Final?

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_CodeStructureTest.java [hunk 3]
```diff
@@ -1,9 +1,31 @@
 package uk.gov.justice.services.adapters.rest.generator;
 
+import org.junit.Test;
```
Comment (purple52) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_CodeStructureTest.java]: Imports have rearranged on this file too

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_POSTMethodBodyTest.java [hunk 4]
```diff
@@ -1,10 +1,22 @@
 package uk.gov.justice.services.adapters.rest.generator;
 
+import org.apache.cxf.jaxrs.impl.tl.ThreadLocalHttpHeaders;
```
Comment (purple52) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_POSTMethodBodyTest.java]: Import ordering again

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java [hunk 5]
```diff
@@ -1,40 +1,42 @@
 package uk.gov.justice.services.example.cakeshop.it;
 
-import static com.jayway.awaitility.Awaitility.await;
```
Comment (purple52) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: Import ordering

File: adapters/adapters-test-utils/src/main/java/uk/gov/justice/services/adapters/test/utils/dispatcher/SynchronousRecordingDispatcher.java (hunk 1 above)
Comment (mapingo) [inline adapters/adapters-test-utils/src/main/java/uk/gov/justice/services/adapters/test/utils/dispatcher/SynchronousRecordingDispatcher.java]: Fixed

File: adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/application/CommonProviders.java (hunk 2 above)
Comment (mapingo) [inline adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/application/CommonProviders.java]: Fixed

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_CodeStructureTest.java (hunk 3 above)
Comment (mapingo) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_CodeStructureTest.java]: Fixed

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_POSTMethodBodyTest.java (hunk 4 above)
Comment (mapingo) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_POSTMethodBodyTest.java]: Fixed

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java (hunk 5 above)
Comment (mapingo) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: Fixed

**Pull request merged**